src/eppo-client.ts

```diff
@@ -13,50 +15,58 @@ export interface IEppoClient {
   /**
    * Maps a subject to a variation for a given experiment.
    *
-   * @param subject an entity ID, e.g. userId
+   * @param subjectKey an identifier of the experiment subject, for example a user ID.
    * @param experimentKey experiment identifier
+   * @param subjectAttributes optional attributes associated with the subject, for example name and email.
+   * The subject attributes are used for evaluating any targeting rules tied to the experiment.
    * @returns a variation value if the subject is part of the experiment sample, otherwise null
    * @public
    */
-  getAssignment(subject: string, experimentKey: string): string;
-
-  /**
-   * Returns a Promise that resolves once the client polling process has started.
-   * @public
-   */
-  waitForInitialization: () => Promise<void>;
```

Comment:
I removed this `waitForInitialization` function and just made the `init` method async since that seems simpler.